src/main/java/com/example/petbuddybackend/service/offer/OfferService.java

```diff
@@ -0,0 +1,278 @@
+package com.example.petbuddybackend.service.offer;
+
+import com.example.petbuddybackend.dto.offer.OfferConfigurationDTO;
+import com.example.petbuddybackend.dto.offer.OfferDTO;
+import com.example.petbuddybackend.entity.amenity.AnimalAmenity;
+import com.example.petbuddybackend.entity.animal.Animal;
+import com.example.petbuddybackend.entity.animal.AnimalAttribute;
+import com.example.petbuddybackend.entity.offer.Offer;
+import com.example.petbuddybackend.entity.offer.OfferConfiguration;
+import com.example.petbuddybackend.entity.offer.OfferOption;
+import com.example.petbuddybackend.entity.user.Caretaker;
+import com.example.petbuddybackend.repository.amenity.AnimalAmenityRepository;
+import com.example.petbuddybackend.repository.animal.AnimalAttributeRepository;
+import com.example.petbuddybackend.repository.animal.AnimalRepository;
+import com.example.petbuddybackend.repository.offer.OfferConfigurationRepository;
+import com.example.petbuddybackend.repository.offer.OfferOptionRepository;
+import com.example.petbuddybackend.repository.offer.OfferRepository;
+import com.example.petbuddybackend.repository.user.CaretakerRepository;
+import com.example.petbuddybackend.service.mapper.OfferConfigurationMapper;
+import com.example.petbuddybackend.service.mapper.OfferMapper;
+import com.example.petbuddybackend.utils.exception.throweable.AnimalAmenityAlreadySelectedInOfferException;
+import com.example.petbuddybackend.utils.exception.throweable.NotFoundException;
+import com.example.petbuddybackend.utils.exception.throweable.OfferConfigurationAlreadyExistsException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.keycloak.common.util.CollectionUtil;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import java.text.MessageFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class OfferService {
+
+    private final CaretakerRepository caretakerRepository;
+    private final OfferRepository offerRepository;
+    private final OfferConfigurationRepository offerConfigurationRepository;
+    private final OfferOptionRepository offerOptionRepository;
+    private final AnimalRepository animalRepository;
+    private final AnimalAttributeRepository animalAttributeRepository;
+    private final AnimalAmenityRepository animalAmenityRepository;
+    private final OfferMapper offerMapper = OfferMapper.INSTANCE;
+    private final OfferConfigurationMapper offerConfigurationMapper = OfferConfigurationMapper.INSTANCE;
+
+    public OfferDTO addOrEditOffer(OfferDTO offer, String caretakerEmail) {
```

Comment:
Rozdzieliłbym OfferDTO na OfferRequest i OfferResponse bo OfferDTO ma pola, które nie są potrzebne w requeście (np. id)

Reply:
wiem, ale front moze ich nie podac i bedzie wszystko dzialalo, a obsluzymy jedną klasą. Myśle, ze to nie jest jakis duzy problem ze cos zostanie puste